src/Application.ts

```diff
@@ -0,0 +1,32 @@
+import { ExpressServer } from './ExpressServer';
+import { GithubExtractor } from './GithubExtractor';
+import { Logger } from './Logger';
+
+export class Application {
+
+    constructor(private readonly timeBetweenExtraction: number, private readonly logger: Logger, private readonly githubExtractor: GithubExtractor, private readonly expressServer: ExpressServer) {
+        this.expressServer.start();
+    }
+
+    stop(cb: () => any): void {
+        this.expressServer.close(cb);
+    }
+
+    startExtractionProcess = async () => {
```

Comment:
Si tu y vas avec cette syntaxe là (j'imagine que c'est pour garder le this dans le setTimeout):

```suggestion
    readonly startExtractionProcess = async () => {
```